pkg/sql/schemachanger/scexec/scmutationexec/index.go

```diff
@@ -350,6 +350,29 @@ func (m *visitor) SetIndexName(ctx context.Context, op scop.SetIndexName) error
 	return nil
 }
 
+func (m *visitor) SetConstraintName(ctx context.Context, op scop.SetConstraintName) error {
+	tbl, err := m.checkOutTable(ctx, op.TableID)
+	if err != nil {
+		return err
+	}
+	constraint, err := tbl.FindConstraintWithID(op.ConstraintID)
+	if err != nil {
+		return err
+	}
+	if constraint.AsUniqueWithIndex() != nil {
+		constraint.AsUniqueWithIndex().IndexDesc().Name = op.Name
```

Comment:
We have to be careful with index-backed constraints, because their names are already modelled by the `scpb.IndexName` element. I'm not keen on having different elements represent the same thing so can you make `ConstraintName` really represent non-index-backed-constraint names? And in this line of code, return an assertion error.